### DISCUSSION TITLE
How do I create a paginated blog listing page, with optional category url param?

### DISCUSSION THREAD

**Discussion by toddpadwick** [nuxt/nuxt#9560] [Questions (Nuxt 2)]
I have a blog listing page which I need to paginate, and included optional params. The category should also be in the blog article URL as well. so any one of the following URLs could be possible:

`/blog` - would show the first page of blog articles
`/blog/p2` - would show the second page of blog articles
`/blog/news` - would show the first page of blog articles filtered by the 'news' category
`/blog/news/p2` - would show the second page of blog articles filtered by the 'news' category
`/blog/news/checkout-our-new-website` - would show an article page itself

Any directory structure I try causes any one of the following issues:

- The page param gets confused with the article slug or category slug so rather than showing a page number, it tries to load the article template
- I have to repeat the index.vue file for both `/blog` and `/blog/_category`

So I can't find a way to make this possible. I'm sure it's a very common scenario for blogs so I'm sure there must be an easy solution for this. Can anyone help?

**Caveats:**
- We cannot create multiple index.vue components for `blog`, `blog/p2` and `blog/news` etc (even if we call a single component into them) because that would result In a full page refresh rather than a smooth transition of just the listed articles. When a new category is added, or when next page is clicked, we only want the list of articles to update, not the whole page. our solution must allow for a single index.vue called for all three scenarios.

**Comment by bdrtsky** [comment]
If you want to keep exact same URL's scheme, consider such structure

```
blog
  index.vue // 1st page or anything
  _category-or-pagination // a directory, where you should detect if this is category or pagination
    _slug-or-pagination // same, but check if it's an article or pagination
```

Then depending on what you detected show appropriate component.

To make it simpler to detect if it's pagination, I would probably changed `p2` to `2`.

Another way to tackle this, I've made a paginated URL like this `/blog/page/2` instead. This way you can scope pagination in static route.

Would love to hear another way to handle this use case 🙌

**Reply by toddpadwick** [reply]
Thanks @bdrtsky - this makes sense and is definitely one step closer. However I do see a potential issue: Because the paginated index.vue and none paginated index.vue are different pages (even if they pull in the same component), when you click next page, it won't be a smooth transition only affecting the chunked listed articles, it will end up being a full-page change which takes away the benefit of it being an SPA. It also means when you filter by category, it again will be a full page change rather than just filtering the articles as you'd expect.
Do you see what I mean?

**Reply by ghost** [reply]
hi @bdrtsky, what about router modules? can he use "router" or router-extras module to define URLs and use only one vue page for pagination and filtering? because I have the same problem.
thanks.

**Reply by toddpadwick** [reply]
Hey @Nik96i ... Did you get anywhere with this problem?
Doesn't seem like I'm getting much traction on this thread from Nuxt, and I'm struggling to find a way of achieving it anywhere :(

**Reply by ghost** [reply]
yes, this will help you:
[https://stackoverflow.com/questions/68604730/vue-router-define-pretty-url-for-filter-page-with-pagination](url)
let me know if you have any questions. and remember the order of routes you define is important.

**Reply by toddpadwick** [reply]
excellent, thanks @Nik96i . i had trouble with the order and ended up having to prepend (`unshift`) a few to routes rather than appending (`push`) them which worked, but feels a bit hacky.
